src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -519,6 +519,204 @@ public static async ValueTask<FormulaValue> FilterTable(EvalVisitor runner, Eval
+                            var rightRenamed = (RecordValue)await RenameColumns(runner, context, IRContext.NotInSource(rightRow.Value.Type), BuildRenamingArgs(rightRow.Value, rightRenaming)).ConfigureAwait(false);
+
+                            innerRows.Add(DValue<RecordValue>.Of(FormulaValue.NewRecordFromFields(leftRenamed.OriginalFields.Concat(rightRenamed.OriginalFields).ToArray())));
+                        }
+                        else if (result.IsError)
+                        {
+                            // Lambda evaluation resulted in an error. Include the error.
+                            innerRows.Add(result);
+                        }
+
+                        innerDict[leftRow] = true;
+                        innerDict[rightRow] = true;
+
+                        if (outerLeft)
+                        {
+                            if (outerDict.ContainsKey(leftRow))
+                            {
+                                outerDict.Remove(leftRow);
+                            }
+                        }
+
+                        if (outerRight)
+                        {
+                            if (outerDict.ContainsKey(rightRow))
+                            {
+                                outerDict.Remove(rightRow);
+                            }
+                        }
+
+                        continue;
+                    }
+
+                    if (outerLeft)
+                    {
+                        if (!innerDict.ContainsKey(leftRow))
+                        {
+                            outerDict[leftRow] = true;
+                        }
+                    }
+
+                    if (outerRight)
+                    {
+                        if (!innerDict.ContainsKey(rightRow))
+                        {
+                            outerDict[rightRow] = true;
+                        }
+                    }
+                }
+            }
+
+            return innerRows.Concat(outerDict.Keys).ToArray();
+        }
+
+        private static FormulaValue[] BuildRenamingArgs(RecordValue row, RecordValue renaming)
+        {
+            var ret = new List<FormulaValue>() { row };
+
+            foreach (var field in renaming.OriginalFields)
+            {
+                ret.Add(FormulaValue.New(field.Name));
+                ret.Add((StringValue)field.Value);
+            }
+
+            return ret.ToArray();
+        }
+
+        private static async Task<DValue<RecordValue>> LazyCheckPredicateAsync(
+           EvalVisitor runner,
+           EvalVisitorContext context,
+           DValue<RecordValue> leftRow,
+           DValue<RecordValue> rightRow,
+           LambdaFormulaValue predicate)
+        {
+            var scopeValue = FormulaValue.NewRecordFromFields(
+                new NamedValue("LeftRecord", leftRow.Value), 

```

Comment:
>LeftRecord [](http://example.com/codeflow?start=32&length=10)

nit - can we share this identifer with Core so that's the literal is only used once?